tests/lava/proc/io/test_injector.py

```diff
@@ -0,0 +1,537 @@
+# Copyright (C) 2021-22 Intel Corporation
```

Comment:
Please document what in the individual tests is the most important thing. You said that this is the order in which certain events happen. Please clarify this in the docstring of each test where this applies, also for the extractor.

Reply:
Refactored some of the tests to reduce duplication, and added explanation about the most important things in the relevant places.